python/GafferUI/_StyleSheet.py

```diff
@@ -1638,6 +1638,20 @@ def styleColor( key ) :
 		background: $background;
 	}
 
+	QFrame[gafferClass="GafferImageUI.ImageViewUI._CompareImageWidget"] #menuDownArrow {
```

Comment:
I think the reason the original styling isn't getting inherited is that it is based on an exact match for the most derived class in Gaffer. Instead you can do a match against the base classes too, as documented here : https://github.com/GafferHQ/gaffer/blob/main/python/GafferUI/_StyleSheet.py#L203.